AY2021S2-CS2103-T14-1/tp
# Add Undo/Redo commands

## What this does
This PR
1. adds undo/redo capabilities to SunRez
2. adds GUI menu items for undo/redo that enable/disable based on the state of SunRez
3. adds keyboard shortcuts for undo/redo (`CTRL + Z`/`CMD + Z` and `CTRL + SHIFT + Z`/`CMD + SHIFT + Z` respectively)
4. adds UG entries documenting these features

Closes #91 
Closes #92 

## How to test
1. Launch SunRez.
2. Check the `Edit` menu item. The sub-menu items `Undo` and `Redo` should both be disabled.
2. Perform some state-changing commands, but not `alias` (not supported yet).
3. Check the `Edit` menu item again. `Undo` should be enabled now.
3. Execute the `undo` command. The command should be undone.
4. Check the `Edit` menu item again. `Redo` should be enabled now.
4. Execute the `redo` command. The command just undone should be re-executed.
5. Press `CTRL + Z` (Windows, Linux) or `CMD + Z` (Mac) or try the relevant menu options at the top of the SunRez window. This should have the same effect as a regular `undo` command, including adding `undo` and `redo` to command history.
6. Repeat steps 2-8 any number of times and in any order. An error message should be shown if `undo` is executed when there are no more commands to undo, or if `redo` is executed when there are no more commands to redo. Otherwise, a success message should be shown.

## What is left to be done
- Integrate alias command
- Write tests
- Update DG
